coordinator/src/node/event_post_processing.rs

```diff
@@ -0,0 +1,47 @@
+use crate::db;
+use crate::node::Node;
+use crate::routing_fee::models::NewRoutingFee;
+use lightning::util::events::Event;
+use tokio::sync::watch;
+
+pub fn spawn(node: Node, mut node_event_receiver: watch::Receiver<Option<Event>>) {
+    tokio::spawn(async move {
+        loop {
+            match node_event_receiver.changed().await {
+                Ok(()) => {
+                    // we only handle event where the fee is set, i.e. events that do not concern a
+                    // force-close scenario
+                    if let Some(Event::PaymentForwarded {
+                        fee_earned_msat: Some(fee_earned_msat),
+                        prev_channel_id,
+                        next_channel_id,
+                        ..
+                    }) = node_event_receiver.borrow().clone()
+                    {
+                        let mut conn = match node.pool.get() {
+                            Ok(conn) => conn,
+                            Err(e) => {
+                                tracing::error!("Failed to connect to database during node event post processing event: {e:#}");
+                                continue;
+                            }
+                        };
+
+                        if let Err(e) = db::routing_fees::insert(
+                            NewRoutingFee {
```

Comment:
Since the struct `NewRoutingFee` is only used here, I think you can remove it and pass in the fields directly into this function.

Reply:
My timebox for addressing this expired :) 
Diesel kept annoying me when I tried to safe the values directly using a tuple. I don't think it's worth investing more time into fighting with Diesel. 

I generally think it's OK to use the same pattern for the tables and accept this bit of boilerplate. If you think it's important I'm happy to address it in a follow up.